rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
Sorry, I had to do some research first. This is definitely a bug and needs to be fixed through out. I'm in the process of auditing `rcl` for this problem.

In the example above `rmw_destroy_waitset(...)` is never run if `-DNDEBUG` is passed to the compiler.